app/controllers/burgers_controller.rb

```diff
@@ -8,8 +8,29 @@ def show
   end
 
   def new
+    @burger = Burger.new
+    @restaurant = Restaurant.find(params[:restaurant_id])
   end
 
   def create
+    @burger = Burger.new(burger_params)
+    @burger.restaurant = Restaurant.find(params[:restaurant_id])
+    if @burger.save
+      redirect_to burger_path(@burger)
+    else
+      flash[:error] = @burger.errors.full_messages.join(". \n")
```

Comment:
This is really not a best practice approach. I would encourage you to either display your error messages using an array and an unordered list element, or else use the Rails `.to_sentence` helper.